subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/AuthenticationSupporter.java

```diff
@@ -82,14 +84,18 @@ public void credentials(Action<? super PasswordCredentials> action) {
         action.execute(getCredentials(credentialsType));
     }
 
+    public void credentials(Provider<Credentials> credentials) {
```

Comment:
If we would like to promote the `repositories { credentials(provider) }` syntax, this method needs to be added to `org.gradle.api.artifacts.repositories.AuthenticationSupported` and overridden here.